### ISSUE TITLE
Image classification errors when testing against training dataset

### ISSUE THREAD

**Issue by varunlakshmanan** [apple/turicreate#1453]
I have trained an image classifier on a dataset of images, but when I test out the mlmodel file on an iOS app using the same images that the image classifier was trained on, several images have incorrect classifications. What's the problem here? I was expecting that it would have a 100% accuracy against images that it was trained on. 

My initial thought is that it has something to do with if the images were perhaps compressed as they were sent to the iOS device, but I'm not sure.

**Comment by TobyRoseman** [comment]
Hi @varunlakshmanan - 

This is expected behavior. The image classifier is not expected to get 100% accuracy on the train set.

To learn more about the image classifier, see the [How it works](https://apple.github.io/turicreate/docs/userguide/image_classifier/how-it-works.html) page in our User Guide. 

I'm going to close this issue, but let me know if you have more questions.

**Comment by varunlakshmanan** [comment]
Thanks for the link. If the classifier isn't expected to display 100% accuracy on the training set, I'm just curious why after training the classifier, the results matrix displayed an accuracy of 97% against the train set. In real use, it seems like the accuracy rate is closer to 80%.

**Comment by TobyRoseman** [comment]
On average the model will always preform worse on examples that it was not trained with. I assume that's what you mean by "real use". That is also expected.

**Comment by varunlakshmanan** [comment]
Sorry, I think I wasn't clear. By "real use", I meant uploading the same training images from an iOS device (not using completely new images). I was just interested in why there's a discrepancy between the accuracy rate output and the actual accuracy rate when testing on the training set.

**Comment by TobyRoseman** [comment]
There shouldn't be a discrepancy. Can you paste the entire output of your call to `image_classifier.create(...)`.

**Comment by varunlakshmanan** [comment]
This is the output:

![image](https://user-images.githubusercontent.com/17592868/53377051-9013b400-392e-11e9-8e9e-b58c7f233b22.png)
![image](https://user-images.githubusercontent.com/17592868/53377070-a15cc080-392e-11e9-96c3-b556908730f6.png)

**Comment by varunlakshmanan** [comment]
@TobyRoseman This is the output:

![image](https://user-images.githubusercontent.com/17592868/53377051-9013b400-392e-11e9-8e9e-b58c7f233b22.png)
![image](https://user-images.githubusercontent.com/17592868/53377070-a15cc080-392e-11e9-96c3-b556908730f6.png)

**Comment by TobyRoseman** [comment]
@varunlakshmanan - it's still not clear to what the problem is here.

You're training a Turi Create image classifier. Then exporting that model to Core ML. Then checking the accuracy on the training dataset with the Core ML model. There is a big discrepancy between that accuracy and the accuracy reported during training. Is that a correct summary of the problem?

I'm not seeing any Core ML output in your screenshots.

**Comment by TobyRoseman** [comment]
I'm going to close this issue due to inactivity. Feel free to reopen with the requested information.
